sparta/src/Report.cpp

```diff
@@ -445,24 +445,24 @@ class ReportFileParserYAML
                                 }
                             }
                             else{
-                                statistics::expression::Expression expr(assoc_key, n);
+                                Report* const r = report_map_.at(scope.uid);
+                                statistics::expression::Expression expr(assoc_key, n, r->getStatistics());
                                 StatisticInstance si(std::move(expr));
                                 si.setContext(n);
                                 std::string full_name = value;
                                 auto& captures = scope.second;
-                                Report* const r = report_map_.at(scope.uid);
                                 if(getSubstituteForStatName(full_name, n, captures)){
                                     r->add(si, full_name);
                                 }
```

Comment:
This code appears multiple times (407-414, 436-443, 460-468). It might be worth factoring out into a function.

Reply:
Yeah, I noticed that too.  One of those -- "more work than I wanted to invest" but since I'm under the hood...

Reply:
Done